### ISSUE TITLE
Metavariables for phase control and registration need documentation

### ISSUE THREAD

**Issue by nilsvu** [sxs-collaboration/spectre#3266]
# Bug reports:

### Expected behavior:

From reading the metavariables, e.g. for `EvolveScalarWave`, I understand which type aliases must be present and will be retrieved from somewhere further down in the code, and which are just there for convenience. Also, for the required type aliases I can find documentation to understand what they do. Ideally I find this documentation by following "breadcrumbs" in the code.

### Current behavior:

The metavars have type aliases related to phase control and registration that are not documented (or if docs exists, I don't find them from reading through the metavars. I have also tried searching in `*.md` files with no success.). Here's the ones I noticed: `phase_changes`, `initialize_phase_change_decision_data`, `phase_change_tags_and_combines_list`, `dg_registration_list`, `registration_list`.

### Suggested solution:

My suggestion is to follow the example of the `factory_creation` metavars for documenting these items.

**Comment by nilsvu** [comment]
@moxcodes could you take this on when you find the time? I'm quite open for suggestions if you have ideas how to structure this sort of documentation. I like the way we have pioneered this for the `factory_creation` metavars though.

**Comment by moxcodes** [comment]
Yep, it's a thing on my to-do list to write some documentation for phase control infrastructure generally (likely for the dev-guide), but I also think it would be valuable to follow the sub-struct model for grouping and documenting the metavariables options specifically.

**Comment by nilsvu** [comment]
Ok great, thanks!
